addons/account/models/account_move.py

```diff
@@ -2685,6 +2685,9 @@ def _post(self, soft=True):
             if move.is_invoice(include_receipts=True) and float_compare(move.amount_total, 0.0, precision_rounding=move.currency_id.rounding) < 0:
                 raise UserError(_("You cannot validate an invoice with a negative total amount. You should create a credit note instead. Use the action menu to transform it into a credit note or refund."))
 
+            if any(line.account_id.deprecated for line in move.line_ids):
+                raise UserError(_("A line of this invoice is using a deprecated account, you cannot post it."))
```

Comment:
Hello @vin-odoo 

I think we should use move_type's label instaed of `invoice`.

```suggestion
                type_name_mapping = {k: v for k, v in
                             self._fields['move_type']._description_selection(self.env)}
                move_type_name = type_name_mapping[move.move_type]
                raise UserError(_("A line of this %s is using a deprecated account, you cannot post it."), % move_type_name)
```

Reply:
Thank you for the feedback. We decided with Benj to go with the more generic "move".